osu.Game/Graphics/UserInterface/ScreenBreadcrumbControl.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Linq;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Screens;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class ScreenBreadcrumbControl : ScreenBreadcrumbControl<Screen>
+    {
+    }
+
+    public class ScreenBreadcrumbControl<T> : BreadcrumbControl<T> where T : Screen
+    {
+        private T currentScreen;
+        public T CurrentScreen
```

Comment:
i feel like we want this to be a ctor argument rather than mutable property. would that still work? right now it creates too many weird possibilities since it can be set an any point in execution.